### PR TITLE
`xtask`: `fn utils::run_capturing_stdout`

### LINKED ISSUE #452: Snapshot tests are failing without an error message
Running `cargo xtask test-snapshot` locally results in this output:

```diff
🧪 qemu/snapshot
log (dev)
-0.000000 INFO Hello!
-0.000001 INFO World!
-0.000002 INFO The answer is 42
-0.000003 INFO Hello 42 42!
-0.000004 INFO Hello 256 42 false
-0.000005 INFO 🍕 slice [3, 14]
-0.000006 INFO 🍕 array [3, 14, 1]
-0.000007 INFO float like a butterfly 5.67 5.67
...
```

and so on, for every snapshot test. xtask should display the Cargo/rustc output here instead of just outputting this unhelpful diff.

### PR DESCRIPTION
**Pull request by Urhengulas**
Fix #452

### CODE REVIEW

**Comment by Urhengulas** [comment]
See https://github.com/knurling-rs/defmt/runs/2587873002?check_suite_focus=true#step:5:96 for CI error

**Comment by Urhengulas** [comment]
status:
* handle that `defmt-test` seems to return non-zero exit codes for `#[should_error]` tests
* give an error message better than `Error (see above)`

File: xtask/src/utils.rs [hunk 1]
```diff
@@ -18,10 +19,25 @@ pub fn load_expected_output(name: &str, release_mode: bool) -> anyhow::Result<St
     })
 }
 
-/// Execute the [`Command`] and return the text emitted to `stdout`, if valid UTF-8
+/// Execute the [`Command`]. If success return `stdout`, if failure print `stderr`
```
Comment (Lotterleben) [inline xtask/src/utils.rs:22]: nit:
```suggestion
/// Execute the [`Command`]. If success return `stdout`, if failure print to `stderr`
```

File: xtask/src/utils.rs [hunk 2]
```diff
@@ -18,10 +19,25 @@ pub fn load_expected_output(name: &str, release_mode: bool) -> anyhow::Result<St
...
 pub fn run_capturing_stdout(cmd: &mut Command) -> anyhow::Result<String> {
-    let stdout = cmd.output()?.stdout;
-    Ok(str::from_utf8(&stdout)?.to_string())
+    let output = cmd.output()?;
+    match output.status.success() {
+        true => Ok(str::from_utf8(&output.stdout)?.to_string()),
+        false => {
+            // #[shoud_error]-tests return non-zero code
```
Comment (Lotterleben) [inline xtask/src/utils.rs:28]: nit:
```suggestion
            // #[should_error]-tests return non-zero code
```

**Review by Lotterleben — APPROVED** [review]

**Comment by Urhengulas** [comment]
Thanks for the review @Lotterleben! 😊

**Pull request merged**